src/ansys/sound/core/sound_composer/source_harmonics_two_parameters.py

```diff
@@ -0,0 +1,400 @@
+                "Specified RPM source control object must be of type ``SourceControlTime``."
+            )
+        self.__control_rpm = control
+
+    @property
+    def control2(self) -> SourceControlTime:
+        """Second source control for harmonics source with two parameters.
+
+        Contains the second control parameter values over time.
+        """
+        return self.__control2
+
+    @control2.setter
+    def control2(self, control: SourceControlTime):
+        """Set the source control."""
+        if not (isinstance(control, SourceControlTime) or control is None):
+            raise PyAnsysSoundException(
+                "Specified second source control object must be of type ``SourceControlTime``."
+            )
+        self.__control2 = control
+
+    @property
+    def source_harmonics_two_parameters(self) -> FieldsContainer:
+        """Source data for harmonics source with two parameters.
+
+        The harmonics source with two parameters data consists of a series of orders whose levels
+        depend on the values of two parameters, the first of which is always RPM.
+        """
+        return self.__source_harmonics_two_parameters
+
+    @source_harmonics_two_parameters.setter
+    def source_harmonics_two_parameters(self, source: FieldsContainer):
+        """Set the harmonics source with two parameters data, from a DPF fields container."""
+        if source is not None:
+            if not isinstance(source, FieldsContainer):
+                raise PyAnsysSoundException(
+                    "Specified harmonics source with two parameters must be provided as a DPF "
+                    "fields container."
+                )
+
+            if len(source) < 1:
+                raise PyAnsysSoundException(
+                    "Specified harmonics source with two parameters must contain at least one "
+                    "order."
+                )
+
+            for spectrum in source:
+                if len(spectrum.data) < 1:
+                    raise PyAnsysSoundException(
+                        "Each order in the specified harmonics source with two parameters must "
+                        "contain at least one element."
+                    )
+
+            support_data = source.get_support("control_parameter_1")
+            support_properties = support_data.available_field_supported_properties()
+            support1_values = support_data.field_support_by_property(support_properties[0])
+            support_data = source.get_support("control_parameter_2")
+            support_properties = support_data.available_field_supported_properties()
+            support2_values = support_data.field_support_by_property(support_properties[0])
+            if len(support1_values) != len(source) or len(support2_values) != len(source):
+                raise PyAnsysSoundException(
+                    "Harmonics source with two parameters must contain as many order levels as "
```

Comment:
Harmonics source with two parameters must contain as many set of order levels as "
                    "the number of values in both associated control parameters (in the provided "
                    "DPF fields container, the number of fields should be the same as the number "
                    "of values in both fields container supports)

Reply:
done